build.sbt

```
@@ -73,7 +73,7 @@ lazy val compat = MultiScalaCrossProject(JSPlatform, JVMPlatform)(
       }
     )
     .jvmSettings(
-      OsgiKeys.exportPackage := Seq(s"scala.collection.compat.*;version=${version.value}"),
+      OsgiKeys.exportPackage := Seq(s"scala.collection.compat.*;scala.jdk.*;version=${version.value}"),
```

Is there a reason for this not to be the following?

```suggestion
      OsgiKeys.exportPackage := Seq(s"*;version=${version.value}"),
```

---

I guess the intent of the OSGi proponent was to be explicit about what this "exports"? It might be good to use `*` until we actually do integration testing using the resulting bundle.

---

It should be OK. The packages get expanded anyway.

---

Actually no, it is not OK. Using `*` will export *all* packages that are visible, including all used `scala` subpackages. We have to be more precise and only specify the ones that are really part of this module.

---

the module should export `scala.jdk` only in the 2.11 / 2.12 versions, something like

```
      OsgiKeys.exportPackage := {
        if (scalaVersion.value.startsWith("2.13."))
          Seq(s"scala.collection.compat.*;version=${version.value}")
        else
          Seq(s"scala.collection.compat.*;version=${version.value},scala.jdk.*;version=${version.value}")
      },
```